ratelimit_test.go

```diff
@@ -7,11 +7,22 @@ import (
 
 	"go.uber.org/atomic"
 
-	"github.com/benbjohnson/clock"
 	"github.com/stretchr/testify/assert"
 )
 
+const advanceBackoffDuration = 5 * time.Millisecond
```

Comment:
Do you know why we need this? Is the `gosched` not enough?

Reply:
You can notice that `gosched` calls happen under lock. It means that all goroutines that would like to get the current time from this clock or register some additional timer won't be able to do so, even if we run these gosched calls. In our case, it means that most "startTaking" goroutines will never work while we step through timers, but they need the most time resource in our tests run.

So we actually need both types of pauses between runs. One is using `gosched` under lock for all goroutines that were waiting for the timer to click and are not blocked on mutex of this clock. 
The other one is for all goroutines that are blocked on mutex of this clock and need to make some progress between timer advances.